python/setup.py

```diff
@@ -68,18 +70,21 @@ def run(self):
     shutil.copy(agent_proto_source_path, agent_proto_destination_path)
 
     protoc_command_parts = [
+        __file__,
         f"-I{build_lib_path}",
         f"--python_out={build_lib_path}",
         f"--grpc_python_out={build_lib_path}",
-        agent_proto_destination_path
+        str(agent_proto_destination_path),
     ]
-    # Instead of `self.spawn`, this should be runnable directly as
-    # `grpc_tools.protoc.main(protoc_command_parts)`, but that seems to fail
-    # on MacOS for some reason, most likely because of the lack of explicit
-    # `protoc.py` included by the script-version of `protoc`.
-    self.spawn([
-        "python", "-m", "grpc_tools.protoc", *protoc_command_parts
-    ])
+
+    protoc_returncode = protoc.main(protoc_command_parts)
+
+    if protoc_returncode != 0:
+      raise subprocess.CalledProcessError(
+        returncode=protoc_returncode,
+        cmd=f"`protoc.main({protoc_command_parts})`",
+      )
```

Comment:
Not sure if this is the right error here.

Reply:
It's a bit odd, but it makes sense to try and have the same error type as the other bits in the file.